spring-cloud-azure-feature-management/src/main/java/com/microsoft/azure/spring/cloud/feature/manager/entities/FeatureSet.java

```diff
@@ -43,20 +50,46 @@ public FeatureSet() {
     public void setFeatureManagement(HashMap<String, Object> featureManagement) {
         setFeatures(featureManagement);
     }
-    
+
     public void addFeature(Feature feature) {
-        featureManagement.put(feature.getId(), feature);
+        featureManagement.put(feature.getKey(), feature);
     }
-    
+
     public void setFeatures(HashMap<String, Object> features) {
         if (features == null) {
             return;
         }
-        for (String key: features.keySet()) {
-            if (features.get(key) instanceof Boolean) {
-                onOff.put(key, (Boolean) features.get(key));
+        for (String key : features.keySet()) {
+            convertFeature(features, key, "");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void convertFeature(HashMap<String, Object> features, String key, String combined) {
```

Comment:
convertFeature name is somehow confusing, since it is updating onoff set or/and updating featureManagement set.